### PR TITLE
Fix 1-frame blips on obstructed items

### PR DESCRIPTION
**Pull request by Buurazu**
This should properly fix 1-frame false positives in the obstruction system

There were two issues that caused both obstructed items to be shown and non-obstructed items to be hidden for 1 frame:
On first frame, EID.hasValidWalkingpath will always be true, because the rendering happens after the pathfinding update
On second frame, EID.hasValidWalkingpath will always be false, because the pathfinder entity was just spawned and isn't in position yet

So, these two frames should be ignored entirely. I added a variable to that check to make these two frames get ignored; pathfindingTo will be nil when the pathfinding update has no description, and only gets set once the pathfinder has spawned and actually pathfinded (or you have flight).

I also had to add the ability to update the pathfinding, to handle issues when moving between an unobstructed and obstructed pickup with no blank frame in between. In that situation, everything is set up already so no issue.

Also:

* Comment out a line that adjusted collectible pedestals on top of each other to not be on top of each other, it doesn't crash or anything to let them be vanilla
* Fix the fixed recipes in the XML (the python script needs to sort the fixed recipe text by pickup ID)
* Hide Bag of Crafting during Cantripped challenge
* Remove the two Tear Heights left because they could be confusingly seen as a buff/debuff despite doing essentially nothing
* Delete the Cube of Meat item pool since I believe it's unused (previous achievement-detection check?) in case it affects Chaos or something

### CODE REVIEW

File: main.lua [hunk 1]
```diff
@@ -573,7 +574,7 @@ function EID:onGameUpdate()
 		local pos = entity.Position
 		for _, otherPos in ipairs(curPositions) do
 			if pos:Distance(otherPos[2]) == 0 then
-				entity.Position = entity.Position + Vector(1,0)
+				--entity.Position = entity.Position + Vector(1,0)
```
Comment (wofsauge) [inline main.lua:577]: Please revert! We need this code for mega chests!

**Review by wofsauge — CHANGES_REQUESTED** [review]

**Comment by Buurazu** [comment]
I managed to get left/right collectible detection on Mega Chests working without moving pedestal positions by using the render offset to break "closest collectible position" ties!

**Comment by Buurazu** [comment]
Fix Tainted Forgotten seeing the wrong effect for Bad Trip / Health Down

**Comment by Buurazu** [comment]
Add False PHD callback using the new XML info

And while testing that, I noticed Golden Pills are a bit messed up...
* Fix Golden Pills being treated as unidentified
* Fix Golden Pills causing a nil error when trying to get their effect name
* Fix Golden Horse Pills spoiling what effect they're going to be

**Comment by Buurazu** [comment]
sorry that it got a bit big, I've tested the Mega Chests and pills thoroughly today and everything should be good to go

don't forget to update the python script to sort recipe ingredients before next game update

**Pull request merged**